go-apps/meep-rnis/server/rnis.go

```diff
@@ -230,13 +230,13 @@ func updateUeData(name string, mnc string, mcc string, cellId string, erabId int
 	//updateDB if changes occur
 	if newEcgi.Plmn.Mnc != oldPlmnMnc || newEcgi.Plmn.Mcc != oldPlmnMcc || newEcgi.CellId != oldCellId {
 
-		//allocating a new erabId if entering a 3GPP environment
-		if oldCellId == "" && cellId != "" {
+		//allocating a new erabId if entering a 4G environment (existence of an erabId)
```

Comment:
The logic below is confusing because there are two conditions checked in the if-else statement + the logic changes (i.e. == -1 vs != -1)

The pseudo-code should look like this

`
if oldErabId == -1{
   if erabIdValid {
      // do something
   }
   else {
      // do something else
   }
}
else{
   if erabIdValid {
      // do something
   }
   else {
      // do something else
   }

}
`

Reply:
fixed using the pseudocode and added comments to be more explicit